File: NEsper.Tests/support/filter/SupportFilterSpecBuilder.cs
///////////////////////////////////////////////////////////////////////////////////////
// Copyright (C) 2007 Esper Team. All rights reserved.                                /
// http://esper.codehaus.org                                                          /
// ---------------------------------------------------------------------------------- /
// The software in this package is published under the terms of the GPL license       /
// a copy of which has been included with this distribution in the license.txt file.  /
///////////////////////////////////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;

using net.esper.events;
using net.esper.filter;
using net.esper.support.events;

namespace net.esper.support.filter
{
	public class SupportFilterSpecBuilder
	{
	    public static FilterSpecCompiled Build(EventType eventType, Object[] objects)
	    {
	        String eventTypeId = SupportEventAdapterService.GetService().GetIdByType(eventType);
	        String eventTypeAlias = SupportEventAdapterService.GetService().GetAliasById(eventTypeId);
	        return new FilterSpecCompiled(eventType, BuildList(objects));
	    }

	    public static FilterSpecCompiled Build(String eventTypeId, String eventTypeAlias, Object[] objects)
	    {
	        EventType eventType = SupportEventAdapterService.GetService().GetTypeById(eventTypeId);
	        return new FilterSpecCompiled(eventType, BuildList(objects));
	    }

	    public static List<FilterSpecParam> BuildList(Object[] objects)
	    {
	        List<FilterSpecParam> filterParams = new List<FilterSpecParam>();

	        int index = 0;
	        while (objects.Length > index)
	        {
	            String propertyName = (String) objects[index++];
	            FilterOperator filterOperator = (FilterOperator) objects[index++];

	            if (!FilterOperatorHelper.IsRangeOperator(filterOperator))
	            {
	                Object filterForConstant = objects[index++];
	                filterParams.Add(new FilterSpecParamConstant(propertyName, filterOperator, filterForConstant));
	            }
	            else
	            {
	                double min = Convert.ToDouble(objects[index++]);
	                double max = Convert.ToDouble(objects[index++]);
	                filterParams.Add(new FilterSpecParamRange(propertyName, filterOperator,
	                        new RangeValueDouble(min),
	                        new RangeValueDouble(max)));
	            }
	        }

	        return filterParams;
	    }
	}

} // End of namespace

File: trunk/esper/src/test/java/net/esper/support/filter/SupportFilterSpecBuilder.java
package net.esper.support.filter;

import net.esper.filter.*;
import net.esper.event.EventType;
import net.esper.support.event.SupportEventAdapterService;

import java.util.List;
import java.util.LinkedList;

public class SupportFilterSpecBuilder
{
    public static FilterSpecCompiled build(EventType eventType, Object[] objects)
    {
        String eventTypeId = SupportEventAdapterService.getService().getIdByType(eventType);
        String eventTypeAlias = SupportEventAdapterService.getService().getAliasById(eventTypeId);
        return new FilterSpecCompiled(eventType, buildList(objects));
    }

    public static FilterSpecCompiled build(String eventTypeId, String eventTypeAlias, Object[] objects)
    {
        EventType eventType = SupportEventAdapterService.getService().getTypeById(eventTypeId);
        return new FilterSpecCompiled(eventType, buildList(objects));
    }

    public static List<FilterSpecParam> buildList(Object[] objects)
    {
        List<FilterSpecParam> filterParams = new LinkedList<FilterSpecParam>();

        int index = 0;
        while (objects.length > index)
        {
            String propertyName = (String) objects[index++];
            FilterOperator filterOperator = (FilterOperator) objects[index++];

            if (!(filterOperator.isRangeOperator()))
            {
                Object filterForConstant = objects[index++];
                filterParams.add(new FilterSpecParamConstant(propertyName, filterOperator, filterForConstant));
            }
            else
            {
                double min = ((Number) objects[index++]).doubleValue();
                double max = ((Number) objects[index++]).doubleValue();
                filterParams.add(new FilterSpecParamRange(propertyName, filterOperator,
                        new RangeValueDouble(min),
                        new RangeValueDouble(max)));
            }
        }

        return filterParams;
    }
}

